tasks/src/main/scala/dagr/tasks/gatk/SplitIntervalsForCallingRegions.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.tasks.gatk
+
+import java.nio.file.Files
+
+import dagr.core.tasksystem.SimpleInJvmTask
+import dagr.tasks.DagrDef.{DirPath, PathToFasta, PathToIntervals}
+import dagr.tasks.parallel.PathBasedScatterGatherPipeline.SplitInputPathTask
+import htsjdk.samtools.SAMFileHeader
+import htsjdk.samtools.reference.IndexedFastaSequenceFile
+import htsjdk.samtools.util.{CloserUtil, Interval, IntervalList}
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+
+
+/** Creates a set of interval list files given a sequence dictionary.
+  *
+  * If no intervals are given, it will create one or more interval list files per reference sequence.
+  *
+  * Each input interval will be broken up into non-overlapping regions of at most `regionSize` size.
+  */
+class SplitIntervalsForCallingRegions(ref: PathToFasta,
+                                      intervals: Option[PathToIntervals],
+                                      output: Option[DirPath],
+                                      regionSize: Int = 25000000,
```

Comment:
Two things: 1) this is a _max_ right?  You might emit interval lists that are smaller.  2) It's not really a region size, but a scatter size since you're saying this is the max bases in each scatter?  So maybe `maxBasesPerScatter`?